src/main/java/org/jabref/gui/sharelatex/ShareLatexLoginDialogController.java

```diff
@@ -0,0 +1,80 @@
+package org.jabref.gui.sharelatex;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.PasswordField;
+import javafx.scene.control.TextField;
+
+import org.jabref.gui.AbstractController;
+import org.jabref.logic.sharelatex.SharelatexConnector;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+public class ShareLatexLoginDialogController extends AbstractController<ShareLatexLoginDialogViewModel> {
+
+    private final SharelatexConnector connector = new SharelatexConnector();
+    private final List<ShareLatexProjectViewModel> projects = new ArrayList<>();
+
+    @FXML private Button btnCancel;
+    @FXML private Button btnLogin;
+    @FXML private TextField tbAddress;
+    @FXML private TextField tbUsername;
+    @FXML private PasswordField pfPassword;
+
+    @FXML
+    private void initialize() {
+        viewModel = new ShareLatexLoginDialogViewModel();
+    }
+
+    @FXML
+    private void closeDialog() {
+        getStage().close();
+    }
+
+    @FXML
+    private void signIn() {
+        System.out.println("sign in pressed");
+        System.out.println(tbAddress.getText());
+        System.out.println(tbUsername.getText());
+        System.out.println(pfPassword.getText());
+
+        JsonObject jsonResponse = connector.connectToServer(tbAddress.getText(), tbUsername.getText(),
+                pfPassword.getText());
+
+        if (jsonResponse != null) {
+            if (jsonResponse.has("text")) {
+                System.out.println(jsonResponse.get("text").getAsString());
+            }
+
+            if (jsonResponse.has("projects")) {
+
+                JsonArray projectArray = jsonResponse.get("projects").getAsJsonArray();
+
+                System.out.println(projectArray);
+
+                for (JsonElement elem : projectArray) {
+
+                    String id = elem.getAsJsonObject().get("id").getAsString();
+                    String name = elem.getAsJsonObject().get("name").getAsString();
+                    String lastUpdated = elem.getAsJsonObject().get("lastUpdated").getAsString();
+                    String owner =  elem.getAsJsonObject().get("owner_ref").getAsString();
+                    System.out.println("ID " + id);
+                    System.out.println("Name " + name);
+                    System.out.println("LastUpdated " + lastUpdated);
+                    System.out.println("Owner" + owner);
+
+                    ShareLatexProjectViewModel model = new ShareLatexProjectViewModel(id, name, owner, lastUpdated);
```

Comment:
A few points:
- This code here looks like it mostly belongs to the logic package.
- I would create a logic class that provides a convenient interface for the ShareLatex integration (e.g. methods like `login(username, password)` and `getProjects`). Lets call this class ShareLatexManager for the moment. 
- This manager class can be initialized in https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/gui/DefaultInjector.java#L25 and then injected into each controller that needs access to the sharelatex functionality, for example similar to the journal abbreviation loader https://github.com/JabRef/jabref/blob/ae6d01be46c96ef57e9871dfeb208ab0b7aab295/src/main/java/org/jabref/gui/journals/ManageJournalAbbreviationsController.java#L43
- So you see that in JavaFX you almost never want to pass something directly to a new view. 
- The purpose of view models is to enrich or format the backend data in a nice way to display them directly in the GUI. For example, if one of your sharelatex projects have a property `dateCreated` but you only want to display the year of creation that the `ProjectViewModel` only contains `yearCreated`. But most of the work should really be done in the logic.

Hope this is helpful.